File: 0260-Single Number III/0260-Single Number III.cpp
class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        vector<int> result(2);
        int diff = 0;
        for (int num : nums)
        {
            diff ^= num;
        }
        
        int lastBitMask = (-diff) & diff;
        for (int num : nums)
        {
            if (num & lastBitMask)
            {
                result[0] ^= num;
            }
            else
            {
                result[1] ^= num;
            }
        }
        
        return result;
    }
};

File: 0201-0300/0260-Single Number III/0260-Single Number III.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        diff = 0
        for num in nums:
            diff ^= num
        lastBitMask = (-diff)&diff
        result = [0, 0]
        for num in nums:
            if num & lastBitMask:
                result[0] ^= num
            else:
                result[1] ^= num
        return result
